File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/numberpath/data/NumberPathDocument.java
package com.zwstudio.logicpuzzlesandroid.puzzles.numberpath.data;

import com.zwstudio.logicpuzzlesandroid.common.data.GameDocument;
import com.zwstudio.logicpuzzlesandroid.common.data.MoveProgress;
import com.zwstudio.logicpuzzlesandroid.common.domain.Position;
import com.zwstudio.logicpuzzlesandroid.puzzles.numberpath.domain.NumberPathGame;
import com.zwstudio.logicpuzzlesandroid.puzzles.numberpath.domain.NumberPathGameMove;

import org.androidannotations.annotations.EBean;

@EBean
public class NumberPathDocument extends GameDocument<NumberPathGame, NumberPathGameMove> {
    protected void saveMove(NumberPathGameMove move, MoveProgress rec) {
        rec.row = move.p.row;
        rec.col = move.p.col;
        rec.intValue1 = move.dir;
    }
    public NumberPathGameMove loadMove(MoveProgress rec) {
        return new NumberPathGameMove() {{
            p = new Position(rec.row, rec.col);
            dir = rec.intValue1;
        }};
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/numberpath/NumberPathDocument.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.numberpath

import android.content.Context
import com.zwstudio.logicpuzzlesandroid.common.data.GameDocument
import com.zwstudio.logicpuzzlesandroid.common.data.MoveProgress
import com.zwstudio.logicpuzzlesandroid.common.domain.Position

class NumberPathDocument(context: Context) : GameDocument<NumberPathGameMove>(context) {
    override fun saveMove(move: NumberPathGameMove, rec: MoveProgress) {
        rec.row = move.p.row
        rec.col = move.p.col
        rec.intValue1 = move.dir
    }

    override fun loadMove(rec: MoveProgress) =
        NumberPathGameMove(Position(rec.row, rec.col), rec.intValue1)
}